### PR TITLE
Fs 3475 add feedback export functionality

### PR DESCRIPTION
**Pull request by RamuniN**
https://dluhcdigital.atlassian.net/browse/FS-3475

### Description
- Added API to retrieve all feedback & survey report for given fund/round
- Updated the requirements
- Added unit tests

This API is used in the assessment as "Export feedback link". Clicking on the link will export the feedback data to an excel file.
Assessment PR is [here](https://github.com/communitiesuk/funding-service-design-assessment/pull/422)

### How to test
- Checkout the branch
- Rebuild the container
- Go to swagger & test the API
image

### CODE REVIEW

File: db/queries/reporting/queries.py [hunk 1]
```diff
@@ -58,6 +58,22 @@ def export_json_to_csv(return_data, headers=None):
...
+
+    if not return_data:
+        return output
+
+    with pd.ExcelWriter(output, engine="openpyxl") as writer:
+        for key in return_data.keys():
+            df = pd.DataFrame(return_data[key])
+            df.to_excel(writer, sheet_name=key)
+
+    # seeking was necessary. Python 3.5.2, Flask 0.12.2
```
Comment (srh-sloan) [inline db/queries/reporting/queries.py:72]: Do we need this? We aren't in python 3.5.2

Comment (srh-sloan) [reply db/queries/reporting/queries.py:72]: Also should the api layer be returning stuff formatted in excel? That feels like more of a frontend/consumer thing, and the backend should just supply json....

File: tests/test_all_feedbacks.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return_value=app_sections,
+    )
+    mocker.patch(
+        "db.queries.feedback.queries.get_applications",
+        return_value=applications,
+    )
+
+    result = retrieve_all_feedbacks_and_surveys("test_fund", "test_round", "SUBMITTED")
+    assert "sections_feedback" in result
+    assert "end_of_application_survey_data" in result
```
Comment (srh-sloan) [inline tests/test_all_feedbacks.py:75]: Can you add some checks to the contents of the response please? Could be anything (or nothing) in those items in the response, and there's quite a bit of logic building up that data structure that should be tested.

**Review by srh-sloan — CHANGES_REQUESTED** [review]
Can you add some more detailed tests please, and also not sure about the backend returning excel format

File: db/queries/reporting/queries.py (hunk 1 above)
Comment (RamuniN) [reply db/queries/reporting/queries.py:72]: seeking is necessary from python 3.5.2 so weed need that line of code.

Intially I went with json output from API. But this caused dict keys to scramble and messed up excel fields structure in assessment. There is `OrderedDict` module which can solve this but complicates the implementation. so for now I would to keep it simple & return the excel file from application-store API

File: tests/test_all_feedbacks.py (hunk 2 above)
Comment (RamuniN) [reply tests/test_all_feedbacks.py:77]: Done.

File: tests/test_all_feedbacks.py [hunk 3]
```diff
@@ -73,3 +75,50 @@ def test_retrieve_all_feedbacks_and_surveys(
...
+    # check contents
+    assert result["sections_feedback"][0]["section"] == app_sections[0]["title"]
+    assert (
+        result["sections_feedback"][0]["comment"]
+        == applications[0].feedbacks[0].feedback_json["comment"]
+    )
+    assert (
+        result["sections_feedback"][0]["rating"]
+        == applications[0].feedbacks[0].feedback_json["rating"]
+    )
```
Comment (srh-sloan) [inline tests/test_all_feedbacks.py:88]: 👍

**Review by srh-sloan — APPROVED** [review]
LGTM

**Pull request merged**